pkg/scheduler/objects/application_state.go

```diff
@@ -134,7 +144,15 @@ func NewAppState() *fsm.FSM {
 			},
 			fmt.Sprintf("enter_%s", Completed.String()): func(event *fsm.Event) {
 				metrics.GetSchedulerMetrics().IncTotalApplicationsCompleted()
+				setTimer(completedTimeout, event, deleteApplication)
```

Comment:
We also need to disconnect the app from the queue and vice versa:
```
app.unSetQueue()
```
And in the app have unSetQueue() as:
```
// remove the leaf queue the application runs in, used when completing the app
func (sa *Application) unSetQueue() {
	sa.queue.RemoveApplication()
	sa.Lock()
	defer sa.Unlock()
	sa.queue = nil
}
```
This should be done in combination with the removal of completedApplications in the PartitionContext